domain/abbreviate_test.go

```diff
@@ -7,12 +7,43 @@ import (
 )
 
 func TestMatcher_Match(t *testing.T) {
-	m := Matcher{items: map[string]string{"abbreviation": "abbr"}}
+	m := Matcher{mainWords: map[string]string{"abbreviation": "abbr"}}
 
 	assert.Equal(t, "abbr", m.Match("abbreviation"))
 	assert.Equal(t, "something", m.Match("something"))
 }
 
+func TestMatcher_Prefixes(t *testing.T) {
+
+	m := Matcher{mainWords: map[string]string{"abbreviation": "abbr"},
+		prefixes: map[string]string{"pre": "pr", "anti": "ant"}}
+
+	assert.Equal(t, "prabbr", m.Match("preabbreviation"))
+	assert.Equal(t, "antprabbr", m.Match("antipreabbreviation"))
+}
+
+func TestMatcher_Suffixes(t *testing.T) {
+
+	m := Matcher{mainWords: map[string]string{"censor": "cnsr"},
+		suffixes: map[string]string{"ship": "shp"}}
+
+	assert.Equal(t, "cnsrshp", m.Match("censorship"))
+}
+
+func TestMatcher_PrefixesSuffixes(t *testing.T) {
+
+	m := Matcher{mainWords: map[string]string{"establish": "estblsh"},
+		prefixes: map[string]string{
+			"anti": "ant",
+			"dis":  "ds"},
+		suffixes: map[string]string{
+			"ment":  "mnt",
+			"arian": "arn",
+			"ism":   "sm"}}
+
+	assert.Equal(t, "antdsestblshmntarnsm", m.Match("antidisestablishmentarianism"))
```

Comment:
What would happen is the replacements `dis` and `arian` weren't available? How long would the abbreviation be? Especially since `arian` isn't one of the available suffixes above?

My worry here is that this test is a little too specific to this word.